### PR TITLE
Feature/contributingupdate

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
# Summary

Updating and adding the documentation contribution guide to the main docs repository, as well as updating links to it.

Also updated the root contributiuon doc to point at the main reference in the docs site.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -11,49 +9,18 @@ The following rules must be observed at all times when contributing documentatio
...
-#### Surface Information Hidden in the Code
-
-Being inline with the code allows you to easily look for critical information within it that the user might not know from looking at the API alone.  Take your time to explore inner method calls and platform specific sections of the code.  The time to write the documentation is once you feel you fully understand the code you are documenting.  If you don't feel you understand the code then leave the documentation for someone else to write.
+## Contributing guidelines
 
-#### Documentation Is Referenced Not Read
+For further details for guidance and advice when contributing to the MonoGame Framework Documentation / API repository, please see the official guide:
 
-Remember that the user is searching for an answer for a specific question.  It is your job to predict these questions and provide them clear answers.
+### [MonoGame Documentation official contributing guide](https://monogame.net/articles/contributing)
```
Comment (tomspilman) [inline CONTRIBUTING.md:20]: Should this link exist?

**Comment by SimonDarksideJ** [comment]
OK, guidelines need a little more work based on the feedback from the team doing the API updates

**Review by dellis1972 — DISMISSED** [review]

File: documentation/articles/contributing.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+To keep the documentation packed and readable, each parameter should be contained to a single line, for example:
+
+```csharp
+<summary>
+Creates a new instance of Bgr565.
+</summary>
+<param name="x">The x component</param>
+<param name="y">The y component</param>
+<param name="z">The z component</param>
```
Comment (AristurtleDev) [inline documentation/articles/contributing.md:89]: Should a note or exception be added here when a parameter description exceeds the 120 character width?

For example. the `ContentManager` documentation would currently look like the following

```csharp
        /// <summary>
        /// 
        /// </summary>
        /// <param name="serviceProvider">The service provider that the <see cref="ContentManager"/> should use to locate services.</param>
```

With the guidelines on using the `<see cref>` tag here, this causes the `<pram>` line to extend a total of 140 characters (including the indention due to the namespace and class level indentions

(image attached shows vertical lines at the 80 and 120 character width limits)
image

In these situations would it be ok to split that parameter to a multi line such as 

```csharp
        /// <summary>
        /// 
        /// </summary>
        /// <param name="serviceProvider">
        /// The service provider that the <see cref="ContentManager"/> should use to locate services.
        /// </param>
```

image

**Comment by AristurtleDev** [comment]
I'm not sure if you want to specify guidelines on a few more of the XML tags.  For instance, my review above was in regards to the `<param>` tag and when it exceeds the 120 character width, but this question should also extend to other single line tags such at `<typeparam>`, `<returns>`, and `<exception>`

Another question would be the use of `<see href>` (as opposed to `cref`) to provide a link back in the intellisense to monogame documentation.  

Here is an example of a rather larger one that would be submitted for `ContentManager.LoadLocalized` that covers these questions (and maybe raises more not considered such as the `<para>` tag usage and whether it should be single line, multi-line and/or indented if multi-line like below).

```csharp
        /// <summary>
        /// Loads an asset that has been processed by the Content pipeline.
        /// </summary>
        /// <remarks>
        ///     <para>
        ///         This method attempts to load the asset based on the <see cref="CultureInfo.CurrentCulture"/>
        ///         searching for the asset by name and appending it with with the culture name (e.g. "assetName.en-US")
        ///         or two letter ISO language name (e.g. "assetName.en").  If unsuccessful in finding the asset with
        ///         the culture information prepended, it will fall back to loading the default asset.
        ///     </para>
        ///     <para>
        ///         Before a <see cref="ContentManager"/> can load an asset, you need to add the asset to your
        ///         game project using the steps described in
        ///         <see href="https://monogame.net/articles/content_pipeline/index.html">Adding Content - MonoGame</see>.
        ///     </para>
        /// </remarks>
        /// <typeparam name="T">
        /// The type of asset to load.  <see cref="Effect"/>, <see cref="Model"/>, <see cref="SoundEffect"/>,
        /// <see cref="Song"/>, <see cref="SpriteFont"/>, <see cref="Texture"/>, <see cref="Texture2D"/>,
        /// and <see cref="TextureCube"/> are all supported by default byt he standard Content Pipeline
        /// processor, but additional types may be loaded by extending the processor.
        /// </typeparam>
        /// <param name="assetName">
        /// The asset name, relative to the <see cref="RootDirectory"/>, and not including the .xnb extension.
        /// </param>
        /// <returns>
        /// The loaded asset.  Repeated calls to load the same asset will return the same object instance.
        /// </returns>
        public virtual T LoadLocalized<T> (string assetName)
```

This creates the following intellisense

image

image

File: documentation/articles/contributing.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+```csharp
+    public struct VertexPositionColorNormalTexture : IVertexType
+    {
+        /// <inheritdoc cref="VertexPosition.Position" />
+        public Vector3 Position;
+
+        /// <inheritdoc cref="VertexPositionColor.Color" />
+        public Color Color;
+
+        /// <inheritdoc cref="VertexPositionColorNormalTexture.Normal" />
```
Comment (Maniekko) [inline documentation/articles/contributing.md:116]: The documentation here has the wrong reference - cref points to `VertexPositionColorNormalTexture`, trying to inherit the docline from itself. The correct reference here should be `VertexPositionNormalTexture` without the `Color` part.

Comment (dellis1972) [reply documentation/articles/contributing.md:116]: The old docs will have mistakes in them, if you see them its ok to fix it 😄  👍

**Comment by SimonDarksideJ** [comment]
Thanks for the comments everyone, just grabbing some time to try and update the PR, keep the questions / suggestions coming.

From a recent PR, I need to add few more things:

1. Do not self-reference a `cref` for the API element itself, as it just creates a link to the same page you are writing for.
2. Do not describe something as the thing you are describing, as an example "The Genre class provides information about a genre", which doesn't help anyone if they do not know what a genre is.
3. when using cref tags, make sure to include the display name you want to be presented for the tag, e.g. Instead of `<see cref="Album.Genre"/>` use `<see cref="Album.Genere">Album.Genre</see>`, including the display text to avoid confusion in generation (like HTML display tags)
4. If possible, check the generated output if you can, to ensure the final product of what you have written looks and reads correctly

**Comment by SimonDarksideJ** [comment]
> I'm not sure if you want to specify guidelines on a few more of the XML tags. For instance, my review above was in regards to the `<param>` tag and when it exceeds the 120 character width, but this question should also extend to other single line tags such at `<typeparam>`, `<returns>`, and `<exception>`

I wouldn't want the guidelines to get too obfuscated as we will never hit 100% coverage.  THere has to be some common sense applied when writing the docs, for instance, I thin your suggestion for the docs on the LoadLocalized<T> is fine, as it adds some much needed clarity on a complicated function.

SO these are things that will be discussed on a case-by-case basis with API docs submissions and only add further clarity to the contributing guide if required.

**Comment by SimonDarksideJ** [comment]
All comments addressed and feedback applied

**Review by dellis1972 — APPROVED** [review]

**Pull request merged**